src/descriptors/analysis.rs

```diff
@@ -117,10 +119,12 @@ impl DescKeyChecker {
             if let Some(ref origin) = xpub.origin {
                 let der_paths = xpub.derivation_paths.paths();
                 let first_der_path = der_paths.first().expect("Cannot be empty");
+                let second_der_path = &der_paths[1];
                 // We also rule out xpubs with hardened derivation steps (non-normalized xpubs).
                 let valid = xpub.wildcard == descriptor::Wildcard::Unhardened
                     && der_paths.len() == 2
-                    && first_der_path.into_iter().all(|step| step.is_normal());
+                    && first_der_path.into_iter().all(|step| step.is_normal())
+                    && second_der_path.into_iter().all(|step| step.is_normal());
```

Comment:
Perhaps here you could combine both checks into a single line using `der_paths.into_iter().flatten()`.